pyannote/audio/applications/change_detection.py

```diff
@@ -0,0 +1,411 @@
+    #os.makedirs(store_dir)
+
+    # -- LOAD MODEL --
+    architecture_yml = train_dir + '/architecture.yml'
+    WEIGHTS_H5 = train_dir + '/weights/{epoch:04d}.h5'
+    nb_epoch = 0
+    while True:
+        weights_h5 = WEIGHTS_H5.format(epoch=nb_epoch)
+        if not os.path.isfile(weights_h5):
+                break
+        nb_epoch += 1
+    config_dir = os.path.dirname(os.path.dirname(train_dir))
+    config_yml = config_dir + '/config.yml'
+    with open(config_yml, 'r') as fp:
+            config = yaml.load(fp)
+
+    # -- FEATURE EXTRACTION --
+    feature_extraction_name = config['feature_extraction']['name']
+    features = __import__('pyannote.audio.features',
+                                                fromlist=[feature_extraction_name])
+    FeatureExtraction = getattr(features, feature_extraction_name)
+    feature_extraction = FeatureExtraction(
+            **config['feature_extraction'].get('params', {}))
+
+    # -- SEQUENCE GENERATOR --
+    duration = config['sequences']['duration']
+    step = config['sequences']['step']
+
+    def saveSeg(filepath,filename,chn,segmentation):
+        f = open(filepath,'w')
+        for idx, val in enumerate(segmentation):
+                line = filename+' '+str(idx)+' '+str(chn)+' '+str(int(val[0]*100))+' '+str(int(val[1]*100-val[0]*100))+'\n'
+                f.write(line)
+        f.close()
+
+    def get_aggregation(epoch):
+        weights_h5 = WEIGHTS_H5.format(epoch=epoch)
+        sequence_labeling = SequenceLabeling.from_disk(
+                architecture_yml, weights_h5)
+
+
+        aggregation = SequenceLabelingAggregation(
+                sequence_labeling, feature_extraction,
+                duration=duration, step=step)
+
+        return aggregation
+
+
+    filepath = store_dir+'/'+str(threshold) +'/'
+    os.makedirs(filepath)
+
+    aggregation = get_aggregation(nb_epoch-1)
+
+    predictions = {}
+    for dev_file in getattr(protocol, subset)():
+        uri = dev_file['uri']
+        predictions[uri] = aggregation.apply(dev_file)  
+
+    # initialize peak detection algorithm
+    peak = Peak(alpha=threshold, min_duration=2.5)
+
+    for dev_file in getattr(protocol, subset)():
+        uri = dev_file['uri']
+        prediction = predictions[uri]
+        hypothesis = peak.apply(prediction)
+        filepath = store_dir+'/'+str(threshold) +'/'+uri+'.0.seg'
+        chn = 1
+        saveSeg(filepath,uri,chn,hypothesis)
+
+
+if __name__ == '__main__':
+
+    arguments = docopt(__doc__, version='Speaker change detection')
+    db_yml = os.path.expanduser(arguments['--database'])
+    preprocessors = {'wav': FileFinder(db_yml)}
+
+    if '<database.task.protocol>' in arguments:
+        protocol = arguments['<database.task.protocol>']
+        database_name, task_name, protocol_name = protocol.split('.')
+        database = get_database(database_name, preprocessors=preprocessors)
+        protocol = database.get_protocol(task_name, protocol_name, progress=True)
```

Comment:
Now, there is `pyannote.database.get_protocol` that can be used directly to do everything in one line :)

https://github.com/pyannote/pyannote-database/blob/master/pyannote/database/__init__.py#L102